pkg/vault/metrics.go

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package vault
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type metrics struct {
+	authTotal             prometheus.Counter
+	authLeaseRenewalTotal prometheus.Counter
+}
```

Comment:
This `metrics` struct is quite tiny, we could move these two counters into `Vault` type directly, it will not bloat that struct.